testing-standard.md

```diff
@@ -64,23 +119,70 @@ someModule._action = function(number) { // test with `null`
+
+### Stubs
+If we ***stubbed*** this function, we'd be able to do everything a spy does, but the body of the function code would never actually be called. Instead it gets replaced by the stub, and immediately returns any value we want. So for our test, we could make our `onePlusOne` function return `5`, for example.
+
+Stubs are really the bread and butter of unit testing, because they allow us to fake interactions within our codebase and focus only on testing the code we care about the code defined in the function. This keeps test setup short, keeps tests understandble and makes their results predictable.
+
+Confused? Take the following function:
 
-  return number;                   // should always finish here
+```javascript
+var hoursAlive: function() {
+  var birthHours = getHoursAtBirth();
+  var deathHours = getHoursAtDeath();
+  return deathHours - birthHours;
 };
 ```
 
-#### My function is too complicated, this is too much effort!
+There are three things we should check for in the above function:
+
+ 1. That we retrieve the birthHours.
+ 2. That we retrieve the deathHours.
+ 3. That we correctly calculate the hours alive based on the two.
+
+By stubbing `getHoursAtBirth` and `getHoursAtDeath` we can **a)** check that both of these are called (ticking off points 1 and 2 above), and **b)** ensure that the real versions of `getHoursAtBirth` and `getHoursAtDeath` are never actually called. This is crucially important as we're **not testing that those functions work**. We only care that they've been called and that what we do in the `hoursAlive` function works. 
+
+Other tests will check that `getHoursAtBirth` and `getHoursAtDeath` work, and by building up a suite of tests that ensure each function works as it's supposed to, we can build confidence that our codebase works correctly. It's for this reason that `stubs` are preferred to spies in our unit tests - we don't want to be running code outside of the function we're actually testing during a unit test.
+
+## 1.5 Good Unit Test Practice
+Now that we've covered good folder structure, good test file structure and the purpose of stubs and spies, we've got everything we need to begin writing good tests.
+
+### Within test suites in general:
+
+* Use fixture files with care. They're sensible for large data objects but usually not necessary.
+* Scope variables as close to the tests they're used in as possible (usually at the top of the current `context` block.
 
+### Within `beforeEach` and `afterEach` functions:
+
+* Use `sinon.stub` to limit behaviour with the `beforeEach(callback)` function, `it` block.
+* Restore all stubs after each test with the `afterEach(callback)` function
+* Reference variables for expected results to ease duplication.
+
+### Within `it` blocks:
+
+* Check all return/callback values
+* Check every stub is invoked with all expected parameters
+
+## 1.6 Unit Testing Gotchas
+
+### Hard to test functions
```

Comment:
this should probably be in our server side best practices, about single responsibility principle and keeping functions small and this doc should link to that for it's example with a brief explanation here.